lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/ReceiveContextMessageExceptionTest.cs

```diff
@@ -0,0 +1,156 @@
+    public sealed class ReceiveContextMessageExceptionTest : ReefFunctionalTest
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(ReceiveContextMessageExceptionTest));
+
+        private static readonly string ContextId = "ContextId";
+        private static readonly string ExpectedException = "ExpectedException";
+        private static readonly string ReceivedFailedEvaluator = "ReceivedFailedEvaluator";
+
+        /// <summary>
+        /// This test validates that a failure in the IContextMessageHandler results in a FailedEvaluator.
+        /// </summary>
+        [Fact]
+        public void TestReceiveContextMessageException()
+        {
+            string testFolder = DefaultRuntimeFolder + TestId;
+
+            TestRun(DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<TestReceiveContextMessageExceptionDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<TestReceiveContextMessageExceptionDriver>.Class)
+                .Set(DriverConfiguration.OnEvaluatorFailed, GenericType<TestReceiveContextMessageExceptionDriver>.Class)
+                .Set(DriverConfiguration.OnContextActive, GenericType<TestReceiveContextMessageExceptionDriver>.Class)
+                .Build(),
+                typeof(TestReceiveContextMessageExceptionDriver), 1, "ReceiveContextMessageExceptionTest", "local", testFolder);
+
+            ValidateSuccessForLocalRuntime(0, 0, 1, testFolder);
+            ValidateMessageSuccessfullyLoggedForDriver(ReceivedFailedEvaluator, testFolder);
+            CleanUp(testFolder);
+        }
+
+        private sealed class TestReceiveContextMessageExceptionDriver :
+            IObserver<IDriverStarted>,
+            IObserver<IAllocatedEvaluator>,
+            IObserver<IActiveContext>,
+            IObserver<IFailedEvaluator>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+
+            [Inject]
+            private TestReceiveContextMessageExceptionDriver(IEvaluatorRequestor requestor)
+            {
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                value.SendMessage(Encoding.UTF8.GetBytes("Hello!"));
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                var contextConfig = ContextConfiguration.ConfigurationModule
+                    .Set(ContextConfiguration.Identifier, ContextId)
+                    .Set(ContextConfiguration.OnMessage, GenericType<ReceiveContextMessageExceptionHandler>.Class)
+                    .Build();
+
+                value.SubmitContext(contextConfig);
+            }
+
+            public void OnNext(IFailedEvaluator value)
```

Comment:
So exception throw from the context will result in IFailedEvaluator as the conext is submitted on the Evaulator. In another word, the exception from root context will result in IFailedEvalutor, right? Please add comment.